internal/controllers/machine/machine_controller_noderef.go

```diff
@@ -116,6 +118,17 @@ func (r *Reconciler) reconcileNode(ctx context.Context, cluster *clusterv1.Clust
 		}
 	}
 
+	options := []client.PatchOption{
+		client.FieldOwner("capi-machine"),
+		client.ForceOwnership,
+	}
+	nodeCopy := node.DeepCopy()
+	nodeCopy.Labels = getManagedLabels(machine.Labels)
+	nodePatch := unstructuredNode(nodeCopy)
```

Comment:
```suggestion
	nodePatch := unstructuredNode(node.Name, node.UID, getManagedLabels(machine.Labels))
```
nit. The deepcopy is confusing me a bit. WDYT about rewriting this to avoid the deepcopy. Essentially just calculate the managed labels and then call `unstructuredNode` with exaclty the information we need